Nos/Models/NoteParser.swift

```diff
@@ -148,33 +148,46 @@ struct NoteParser {
     }
 
     private func cleanLinks(
-        in attributedString: AttributedString, 
+        in attributedString: AttributedString,
         tags: [[String]] = []
     ) -> (String, [[String]]) {
-        var mutableAttributedString = attributedString
-        for attributedRun in mutableAttributedString.runs {
-            if let link = attributedRun.attributes.link {
-                var mutableAttributes = attributedRun.attributes
-                mutableAttributes.link = nil
-                mutableAttributedString.replaceSubrange(
-                    attributedRun.range,
-                    with: AttributedString("\(link.absoluteString)", attributes: mutableAttributes)
+        var attributedString = attributedString
+        let runs = attributedString.runs
+        let isLink = { (attributedRun: AttributedString.Runs.Run) in
+            attributedRun.link != nil
+        }
+        var links: [AttributedString.Runs.Run] = runs.filter(isLink).reversed()
+        guard let firstRun = links.popLast(), let link = firstRun.link else {
+            return (String(attributedString.characters), tags)
+        }
+        var attributes = firstRun.attributes
+        var range = firstRun.range
+        while let nextRun = links.popLast(),
+            nextRun.range.lowerBound == firstRun.range.upperBound,
+            nextRun.link == firstRun.link {
+            range = range.lowerBound..<nextRun.range.upperBound
+            attributes.merge(nextRun.attributes)
+        }
+        attributes.link = nil
+        attributedString.replaceSubrange(
+            range,
+            with: AttributedString(
+                "\(link.absoluteString)",
+                attributes: attributes
+            )
+        )
+        if link.scheme == "nostr" {
```

Comment:
shouldn't this still be in a loop of some kind in case there are multiple links? Or I guess it's recursive? I'm having trouble holding in my head everything this function is doing. A headerdoc comment would help.

In any case I'm seeing a problem when doing multiple mentions:

https://github.com/planetary-social/nos/assets/1165004/050f8d25-2d5f-4385-9dba-074799fd3c88